components/merkledb/src/indexes/group.rs

```diff
@@ -106,11 +107,32 @@ where
     }
 }
 
+impl<T, K, I> Group<T, K, I>
+where
+    T: RawAccess,
+    K: BinaryKey + ?Sized,
+    I: FromAccess<T>,
+{
+    /// Returns an iterator over the keys from this group.
+    ///
+    /// **Note:** use this method carefully, because it obtains all available keys for this
+    /// group in `suffixes` method and stores them in `Vec`. This operation can consume a large
+    /// amount of RAM.
+    ///
+    /// **Note:** creating any new indexes while this iterator is in scope may lead to panic.
+    pub fn keys_unchecked(&self) -> impl Iterator<Item = K::Owned> {
+        let indexes_pool = IndexesPool::new(self.access.clone());
+        indexes_pool.suffixes::<K>(self.prefix.name()).into_iter()
```

Comment:
This is incorrect for groups in which `prefix` has an `id_in_group` component. Such groups are impossible to create via `AccessExt`, but are possible to create via `FromAccess` (including via derive macro). You should either provide a generic implementation (as in #1637), or check if the group's prefix has `id_in_group` and panic in this case (or otherwise handle it).